src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -186,31 +189,46 @@ public void setupSchema(List<ColumnMetadata> columns) {
      */
     if (clientBufferParameters.getIsIcebergMode()) {
       for (ColumnDescriptor columnDescriptor : schema.getColumns()) {
-        String columnPath = concatDotPath(columnDescriptor.getPath());
+        String[] path = columnDescriptor.getPath();
+        String columnDotPath = concatDotPath(path);
         PrimitiveType primitiveType = columnDescriptor.getPrimitiveType();
+        boolean isInRepeatedGroup =
+            path.length > 1
+                && schema
+                    .getType(Arrays.copyOf(path, path.length - 1))
+                    .isRepetition(Type.Repetition.REPEATED);
```

Comment:
is it possible to have Repetition.REPEATED on a path BUT its neither a map key, nor a map value, nor a list element.

Can we do an explicit check for those three situations?

Reply:
Added check